### ISSUE TITLE
Basic usage example doesn't compile on rustc 1.9

### ISSUE THREAD

**Issue by ekatsah** [rustless/rustless#63]
When trying to compile the basic usage example, some errors are thrown:
```
src/main.rs:104:35:` 104:55 error: mismatched types:
 expected `rustless::server::status::StatusCode`,
    found `hyper::status::StatusCode`
(expected enum `rustless::server::status::StatusCode`,
    found enum `hyper::status::StatusCode`) [E0308]
src/main.rs:104                 client.set_status(StatusCode::NotFound);
                                                  ^~~~~~~~~~~~~~~~~~~~
src/main.rs:104:35: 104:55 help: run `rustc --explain E0308` to see a detailed explanation
src/main.rs:113:44: 113:59 error: mismatched types:
 expected `Box<valico::json_dsl::coercers::Coercer + Send + Sync + 'static>`,
    found `Box<valico::json_dsl::Coercer + Send + Sync + 'static>`
(expected trait `valico::json_dsl::coercers::Coercer`,
    found trait `valico::json_dsl::Coercer`) [E0308]
src/main.rs:113                     params.req_typed("id", json_dsl::u64())
                                                           ^~~~~~~~~~~~~~~
src/main.rs:113:44: 113:59 help: run `rustc --explain E0308` to see a detailed explanation
src/main.rs:124:53: 124:68 error: mismatched types:
 expected `Box<valico::json_dsl::coercers::Coercer + Send + Sync + 'static>`,
    found `Box<valico::json_dsl::Coercer + Send + Sync + 'static>`
(expected trait `valico::json_dsl::coercers::Coercer`,
    found trait `valico::json_dsl::Coercer`) [E0308]
src/main.rs:124                         params.req_typed("user_id", json_dsl::u64());
                                                                    ^~~~~~~~~~~~~~~
src/main.rs:124:53: 124:68 help: run `rustc --explain E0308` to see a detailed explanation
src/main.rs:125:50: 125:68 error: mismatched types:
 expected `Box<valico::json_dsl::coercers::Coercer + Send + Sync + 'static>`,
    found `Box<valico::json_dsl::Coercer + Send + Sync + 'static>`
(expected trait `valico::json_dsl::coercers::Coercer`,
    found trait `valico::json_dsl::Coercer`) [E0308]
src/main.rs:125                         params.req_typed("name", json_dsl::string())
                                                                 ^~~~~~~~~~~~~~~~~~
src/main.rs:125:50: 125:68 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to 4 previous errors
Could not compile `beam6`.
```

The first one is easy, instead of ```use hyper::status::StatusCode;```, use ```use rustless::server::status::StatusCode;```.
I don't understand how to fix the three other errors. Any idea?

**Comment by s-panferov** [comment]
@ekatsah thanks for your report, I'll take a look today.

**Comment by BjarkeJM** [comment]
Just tried to run the basic example code and I am also experiencing this issue. Did you find a resolution?

**Comment by ptxmac** [comment]
The problem is that the README doesn't specify what version of the dependencies are needed. In this case setting `valico = "0.8"` in Cargo.toml solves the problem.

**Comment by oddRaven** [comment]
Changed to `rustless::server::status::StatusCode` and the code compiles and runs on port 4000, but I still get a 404 Not Found.

>curl -i -H Accept:application/vnd.chat.v1+json -X POST http://localhost:4000/chats/1/users/1
>HTTP/1.1 404 Not Found
>Content-Length: 0
>Date: Sat, 02 Jul 2016 21:15:18 GMT

What is wrong?

**Comment by s-panferov** [comment]
I believe this is fixed in 0.9.0. Please try
